ducktape/cluster/json.py

```diff
@@ -23,11 +23,8 @@ class JsonCluster(Cluster):
     An implementation of Cluster that uses static settings specified in a cluster file.
     """
 
-    def __init__(self, cluster_json=None):
+    def __init__(self, cluster_json, *args, **kwargs):
         super(JsonCluster, self).__init__()
-        if cluster_json is None:
```

Comment:
@lindong28 
We definitely still need to be able to bring up a `JsonCluster` from a file, but I see your point about this overlap. It would be good to sort out what we want the user-facing behavior to be here.

Below is an idea to unify the usage somewhat, what do you think?

example usage:
JsonCluster:
`ducktape --cluster ductape.cluster.json.JsonCluster --cluster-file cluster.json ...`
Behavior:
- load in cluster information from file as before, but user specifies location of file.
- (optionally allow `cluster.json` to continue to be the default location?)

VagrantCluster (the default)
`ducktape --cluster-file cluster.json ...`
Behavior:
- If cluster-file is not specified, gather json data every time as we do today by calling `vagrant ssh-info`
- If cluster-file is specified, but file is not present, call `vagrant ssh-info` and cache this information to cluster-file for future use
- If cluster-file is specified and file is present, load directly from file